src/main/java/org/svydovets/queryLanguage/QueryManager.java

```diff
@@ -0,0 +1,265 @@
+
+    private List<String> joinQueryList = new ArrayList<>();
+
+    public QueryManager(final Class<T> entityType) {
+        this.entityType = entityType;
+    }
+
+    public static <T> QueryManager<T> of(Class<T> entityType) {
+        return new QueryManager<>(entityType);
+    }
+
+    public static <T> QueryManager<T> of(final String query, final Class<T> entityType) {
+        QueryManager<T> queryManager = QueryManager.of(entityType);
+        queryManager.query(query);
+
+        return queryManager;
+    }
+
+    /**
+     * set jql query into QueryManager
+     *
+     * @param query
+     */
+    public void query(final String query) {
+        this.arrQuery = Arrays.asList(query.split(" "));
+
+        initJoinQuery(entityType);
+    }
+
+    /**
+     * set parameters into QueryManager
+     *
+     * @param paramName
+     * @param value
+     */
+    public void setParameters(final String paramName, final Object value) {
+        parameters.put(":" + paramName, value);
+    }
+
+    /**
+     * get parameters into QueryManager
+     *
+     * @return
+     */
+    public Object[] getParameters() {
+        Set<String> keyParameters = parameters.keySet();
+
+        return arrQuery.stream()
+                .filter(field -> keyParameters.contains(replace(field)))
+                .map(key -> parameters.get(replace(key)))
+                .toArray();
+    }
+
+    /**
+     * get entityType from QueryManager
+     *
+     * @return
+     */
+    public Class<T> getEntityType() {
+        return entityType;
+    }
+
+    /**
+     * build native query from QueryManager
+     *
+     * @return
+     */
+    public String toSqlString() {
+        Set<String> keyParams = parameters.keySet();
+        final String tableName = ParameterNameResolver.resolveTableName(entityType);
+        final Map<String, String> columnNameByFieldNameMap = ParameterNameResolver
+                .getColumnNameByFieldNameForColumnFielsMap(entityType);
+        final Set<String> fieldNames = columnNameByFieldNameMap.keySet();
+        final Map<String, String> columnNameByFieldNameForEntityMap = ParameterNameResolver
+                .getColumnNameByFieldNameForEntityFielsMap(entityType);
+        final Set<String> columnNameByFieldNameForEntityKeys = columnNameByFieldNameForEntityMap.keySet();
+        List<String> nativeQueryList = arrQuery.stream()
+                .map(element -> getValueSqlSubqueryString(element, keyParams))
+                .map(element -> {
+                    String columnName = replace(element);
+                    Optional<String> fieldNameOptional = getNameSqlSubqueryString(fieldNames, columnName);
+                    if (fieldNameOptional.isEmpty()) {
+                        return element;
+                    }
+
+                    String nativeColumnName = columnNameByFieldNameMap.get(fieldNameOptional.get());
+
+                    return element.replace(fieldNameOptional.get(), nativeColumnName);
+                })
```

Comment:
We should move this logic into a separate method. This will increase the reliability and flexibility of the code, and besides, it is not a good practice to write code in lambda blocks.